Build fail with o_net.c:175:36: error: ‘current_rules’ may be used uninitialized

On a gentoo system with gcc-8.2.0, the last git code for lepton-eda fail to build with
`make[4]: Entering directory '/var/tmp/portage/sci-electronics/lepton-eda-9999/work/lepton-eda-9999/schematic/src'

...snip...
```
x86_64-pc-linux-gnu-gcc -DLOCALEDIR=\"/usr/share/locale\" -DHAVE_CONFIG_H -I. -I../..  -I../../liblepton/include  -I./../include -I../.. -I/usr/include  -Wall -Werror=uninitialized -std=c99    -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -pthread -I/usr/include/guile/2.0 -march=native -O2 -pipe -c -o lepton_schematic-o_net.o `test -f 'o_net.c' || echo './'`o_net.c
o_net.c: In function ‘o_net_guess_direction’:
o_net.c:175:36: error: ‘current_rules’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
    right = MAX(right, current_rules[1]);
                                    ^
cc1: some warnings being treated as errors
make[4]: *** [Makefile:2011: lepton_schematic-o_net.o] Error 1`
```

--- a/schematic/src/o_net.c
+++ b/schematic/src/o_net.c
@@ -79,7 +79,7 @@ void o_net_reset(GschemToplevel *w_current)
  *  as a bitfield.
  */
 void o_net_guess_direction(GschemToplevel *w_current,
-			   int wx, int wy)
+                           int wx, int wy)
 {
   int up=0, down=0, left=0, right=0;
   int x1, y1, x2, y2;
@@ -107,20 +107,20 @@ void o_net_guess_direction(GschemToplevel *w_current,
       o_current = (OBJECT*) iter2->data;
 
       if ((orientation = geda_net_object_orientation (o_current)) == NEITHER)
-	continue;
+        continue;
 
       switch (o_current->type) {
       case OBJ_NET:
-	current_rules = (int*) net_rules;
-	break;
+        current_rules = (int*) net_rules;
+        break;
       case OBJ_PIN:
-	current_rules = (int*) pin_rules;
-	break;
+        current_rules = (int*) pin_rules;
+        break;
       case OBJ_BUS:
-	current_rules = (int*) bus_rules;
-	break;
+        current_rules = (int*) bus_rules;
+        break;
       default:
-	g_assert_not_reached ();
+        g_assert_not_reached ();
       }
 
       x1 = o_current->line->x[0];
@@ -134,50 +134,50 @@ void o_net_guess_direction(GschemToplevel *w_current,
       ymax = MAX(y1, y2);
 
       if (orientation == HORIZONTAL && wy == y1) {
-	if (wx == xmin) {
-	  up = MAX(up, current_rules[1]);
-	  down = MAX(down, current_rules[1]);
-	  right = MAX(right, current_rules[0]);
-	  left = MAX(left, current_rules[2]);
-	}
-	else if (wx == xmax) {
-	  up = MAX(up, current_rules[1]);
-	  down = MAX(down, current_rules[1]);
-	  right = MAX(right, current_rules[2]);
-	  left = MAX(left, current_rules[0]);
-	}
-	else if (xmin < wx && wx < xmax) {
-	  up = MAX(up, current_rules[1]);
-	  down = MAX(down, current_rules[1]);
-	  right = MAX(right, current_rules[0]);
-	  left = MAX(left, current_rules[0]);
-	}
-	else {
-	  continue;
-	}
+        if (wx == xmin) {
+          up = MAX(up, current_rules[1]);
+          down = MAX(down, current_rules[1]);
+          right = MAX(right, current_rules[0]);
+          left = MAX(left, current_rules[2]);
+        }
+        else if (wx == xmax) {
+          up = MAX(up, current_rules[1]);
+          down = MAX(down, current_rules[1]);
+          right = MAX(right, current_rules[2]);
+          left = MAX(left, current_rules[0]);
+        }
+        else if (xmin < wx && wx < xmax) {
+          up = MAX(up, current_rules[1]);
+          down = MAX(down, current_rules[1]);
+          right = MAX(right, current_rules[0]);
+          left = MAX(left, current_rules[0]);
+        }
+        else {
+          continue;
+        }
       }
       if (orientation == VERTICAL && wx == x1) {
-	if (wy == ymin) {
-	  up = MAX(up, current_rules[0]);
-	  down = MAX(down, current_rules[2]);
-	  right = MAX(right, current_rules[1]);
-	  left = MAX(left, current_rules[1]);
-	}
-	else if (wy == ymax) {
-	  up = MAX(up, current_rules[2]);
-	  down = MAX(down, current_rules[0]);
-	  right = MAX(right, current_rules[1]);
-	  left = MAX(left, current_rules[1]);
-	}
-	else if (ymin < wy && wy < ymax) {
-	  up = MAX(up, current_rules[0]);
-	  down = MAX(down, current_rules[0]);
-	  right = MAX(right, current_rules[1]);
-	  left = MAX(left, current_rules[1]);
-	}
-	else {
-	  continue;
-	}
+        if (wy == ymin) {
+          up = MAX(up, current_rules[0]);
+          down = MAX(down, current_rules[2]);
+          right = MAX(right, current_rules[1]);
+          left = MAX(left, current_rules[1]);
+        }
+        else if (wy == ymax) {
+          up = MAX(up, current_rules[2]);
+          down = MAX(down, current_rules[0]);
+          right = MAX(right, current_rules[1]);
+          left = MAX(left, current_rules[1]);
+        }
+        else if (ymin < wy && wy < ymax) {
+          up = MAX(up, current_rules[0]);
+          down = MAX(down, current_rules[0]);
+          right = MAX(right, current_rules[1]);
+          left = MAX(left, current_rules[1]);
+        }
+        else {
+          continue;
+        }
       }
     }
   }
@@ -190,7 +190,7 @@ void o_net_guess_direction(GschemToplevel *w_current,
 
 #if 0
   printf("o_net_guess_direction: up=%d down=%d left=%d right=%d direction=%d\n",
-	 up, down, left, right, w_current->net_direction);
+         up, down, left, right, w_current->net_direction);
 #endif
   g_list_free (object_list);
 }
@@ -206,7 +206,7 @@ void o_net_guess_direction(GschemToplevel *w_current,
  *  are set to -1.
  */
 void o_net_find_magnetic(GschemToplevel *w_current,
-			 int w_x, int w_y)
+                         int w_x, int w_y)
 {
   int x1, x2, y1, y2, min_x, min_y;
   double mindist, minbest, dist1, dist2;
@@ -239,74 +239,74 @@ void o_net_find_magnetic(GschemToplevel *w_current,
       if (!geda_object_calculate_visible_bounds(page->toplevel, o_current,
                                                 &left, &top, &right, &bottom) ||
           !visible (w_current, left, top, right, bottom))
-	continue; /* skip invisible objects */
+        continue; /* skip invisible objects */
 
       if (o_current->type == OBJ_PIN) {
-	min_x = o_current->line->x[o_current->whichend];
-	min_y = o_current->line->y[o_current->whichend];
+        min_x = o_current->line->x[o_current->whichend];
+        min_y = o_current->line->y[o_current->whichend];
 
-	mindist = hypot(w_x - min_x, w_y - min_y);
-	weight = mindist / MAGNETIC_PIN_WEIGHT;
+        mindist = hypot(w_x - min_x, w_y - min_y);
+        weight = mindist / MAGNETIC_PIN_WEIGHT;
       }
 
       else if (o_current->type == OBJ_NET
-	       || o_current->type == OBJ_BUS) {
-	/* we have 3 possible points to connect:
-	   2 endpoints and 1 midpoint point */
-	x1 = o_current->line->x[0];
-	y1 = o_current->line->y[0];
-	x2 = o_current->line->x[1];
-	y2 = o_current->line->y[1];
-	/* endpoint tests */
-	dist1 = hypot(w_x - x1, w_y - y1);
-	dist2 = hypot(w_x - x2, w_y - y2);
-	if (dist1 < dist2) {
-	  min_x = x1;
-	  min_y = y1;
-	  mindist = dist1;
-	}
-	else {
-	  min_x = x2;
-	  min_y = y2;
-	  mindist = dist2;
-	}
+               || o_current->type == OBJ_BUS) {
+        /* we have 3 possible points to connect:
+           2 endpoints and 1 midpoint point */
+        x1 = o_current->line->x[0];
+        y1 = o_current->line->y[0];
+        x2 = o_current->line->x[1];
+        y2 = o_current->line->y[1];
+        /* endpoint tests */
+        dist1 = hypot(w_x - x1, w_y - y1);
+        dist2 = hypot(w_x - x2, w_y - y2);
+        if (dist1 < dist2) {
+          min_x = x1;
+          min_y = y1;
+          mindist = dist1;
+        }
+        else {
+          min_x = x2;
+          min_y = y2;
+          mindist = dist2;
+        }
 
-	/* midpoint tests */
-	if ((x1 == x2)  /* vertical net */
-	    && ((y1 >= w_y && w_y >= y2)
-		|| (y2 >= w_y && w_y >= y1))) {
-	  if (abs(w_x - x1) < mindist) {
-	    mindist = abs(w_x - x1);
-	    min_x = x1;
-	    min_y = w_y;
-	  }
-	}
-	if ((y1 == y2)  /* horitontal net */
-	    && ((x1 >= w_x && w_x >= x2)
-		|| (x2 >= w_x && w_x >= x1))) {
-	  if (abs(w_y - y1) < mindist) {
-	    mindist = abs(w_y - y1);
-	    min_x = w_x;
-	    min_y = y1;
-	  }
-	}
+        /* midpoint tests */
+        if ((x1 == x2)  /* vertical net */
+            && ((y1 >= w_y && w_y >= y2)
+                || (y2 >= w_y && w_y >= y1))) {
+          if (abs(w_x - x1) < mindist) {
+            mindist = abs(w_x - x1);
+            min_x = x1;
+            min_y = w_y;
+          }
+        }
+        if ((y1 == y2)  /* horitontal net */
+            && ((x1 >= w_x && w_x >= x2)
+                || (x2 >= w_x && w_x >= x1))) {
+          if (abs(w_y - y1) < mindist) {
+            mindist = abs(w_y - y1);
+            min_x = w_x;
+            min_y = y1;
+          }
+        }
 
-	if (o_current->type == OBJ_BUS)
-	  weight = mindist / MAGNETIC_BUS_WEIGHT;
-	else /* OBJ_NET */
-	  weight = mindist / MAGNETIC_NET_WEIGHT;
+        if (o_current->type == OBJ_BUS)
+          weight = mindist / MAGNETIC_BUS_WEIGHT;
+        else /* OBJ_NET */
+          weight = mindist / MAGNETIC_NET_WEIGHT;
       }
       else { /* neither pin nor net or bus */
-	continue;
+        continue;
       }
 
       if (o_magnetic == NULL
-	  || weight < min_weight) {
-	minbest = mindist;
-	min_weight = weight;
-	o_magnetic = o_current;
-	w_current->magnetic_wx = min_x;
-	w_current->magnetic_wy = min_y;
+          || weight < min_weight) {
+        minbest = mindist;
+        min_weight = weight;
+        o_magnetic = o_current;
+        w_current->magnetic_wx = min_x;
+        w_current->magnetic_wy = min_y;
       }
     }
   }
@@ -342,10 +342,10 @@ void o_net_finishmagnetic(GschemToplevel *w_current)
   int primary_zero_length, secondary_zero_length;
 
   primary_zero_length = ((w_current->first_wx == w_current->second_wx)
-			 && (w_current->first_wy == w_current->second_wy));
+                         && (w_current->first_wy == w_current->second_wy));
 
   secondary_zero_length = ((w_current->second_wx == w_current->third_wx)
-			   && (w_current->second_wy == w_current->third_wy));
+                           && (w_current->second_wy == w_current->third_wy));
 
   if (!primary_zero_length && secondary_zero_length) {
     if (w_current->first_wx == w_current->second_wx) {
@@ -532,9 +532,9 @@ void o_net_end(GschemToplevel *w_current, int w_x, int w_y)
                                                    new_net->conn_list);
       if (found_primary_connection)
       {
-      	/* if a net connection is found, reset start point of next net */
-	save_wx = w_current->second_wx;
-	save_wy = w_current->second_wy;
+        /* if a net connection is found, reset start point of next net */
+        save_wx = w_current->second_wx;
+        save_wy = w_current->second_wy;
       }
   }
 
@@ -601,8 +601,8 @@ void o_net_motion (GschemToplevel *w_current, int w_x, int w_y)
   if (magnetic_net_mode) {
     if (w_current->CONTROLKEY) {
       /* set the magnetic marker position to current xy if the
-	 controlkey is pressed. Thus the net will not connect to
-	 the closest net if we finish the net drawing */
+         controlkey is pressed. Thus the net will not connect to
+         the closest net if we finish the net drawing */
       w_current->magnetic_wx = w_x;
       w_current->magnetic_wy = w_y;
     }
@@ -800,16 +800,16 @@ int o_net_add_busrippers(GschemToplevel *w_current, OBJECT *net_obj,
       GList *cl_current2 = net_obj->conn_list;
       done = FALSE;
       while (cl_current2 != NULL && !done) {
-	CONN *tmp_conn = (CONN *) cl_current2->data;
+        CONN *tmp_conn = (CONN *) cl_current2->data;
 
-	if (tmp_conn && tmp_conn->other_object &&
-	    tmp_conn->other_object == bus_object) {
+        if (tmp_conn && tmp_conn->other_object &&
+            tmp_conn->other_object == bus_object) {
 
-	  found_conn = tmp_conn;
-	  done = TRUE;
-	}
+          found_conn = tmp_conn;
+          done = TRUE;
+        }
 
-	cl_current2 = g_list_next(cl_current2);
+        cl_current2 = g_list_next(cl_current2);
       }
 
       if (!found_conn) {
@@ -820,7 +820,7 @@ int o_net_add_busrippers(GschemToplevel *w_current, OBJECT *net_obj,
 
       /* now deal with the found connection */
       if (bus_orientation == HORIZONTAL && net_orientation == VERTICAL) {
-	/* printf("found horiz bus %s %d!\n", bus_object->name,
+        /* printf("found horiz bus %s %d!\n", bus_object->name,
            found_conn->whichone);*/
 
         sign = geda_bus_object_get_ripper_direction (bus_object);
@@ -925,9 +925,9 @@ int o_net_add_busrippers(GschemToplevel *w_current, OBJECT *net_obj,
 
 
       } else if (bus_orientation == VERTICAL &&
-		 net_orientation == HORIZONTAL) {
+                 net_orientation == HORIZONTAL) {
 
-	/* printf("found vert bus %s %d!\n", bus_object->name,
+        /* printf("found vert bus %s %d!\n", bus_object->name,
            found_conn->whichone); */
 
         sign = geda_bus_object_get_ripper_direction (bus_object);
@@ -1040,7 +1040,7 @@ int o_net_add_busrippers(GschemToplevel *w_current, OBJECT *net_obj,
 
     if (w_current->bus_ripper_type == COMP_BUS_RIPPER) {
       GList *symlist =
-	s_clib_search (page->toplevel->bus_ripper_symname, CLIB_EXACT);
+        s_clib_search (page->toplevel->bus_ripper_symname, CLIB_EXACT);
       if (symlist != NULL) {
         rippersym = (CLibSymbol *) symlist->data;
       }

--- a/schematic/src/o_net.c
+++ b/schematic/src/o_net.c
@@ -120,6 +120,7 @@ void o_net_guess_direction(GschemToplevel *w_current,
         current_rules = (int*) bus_rules;
         break;
       default:
+        current_rules = (int*) net_rules;
         g_assert_not_reached ();
       }
 

